content/en/docs/concepts/configuration/secret.md

```diff
@@ -127,15 +132,37 @@ data:
   password: MWYyZDFlMmU2N2Rm
 ```
 
-The data field is a map.  Its keys must consist of alphanumeric characters, '-', '_' or '.'.  The values are arbitrary data, encoded using base64.
-
-Create the secret using [`kubectl create`](/docs/reference/generated/kubectl/kubectl-commands#create):
+Now create the Secret using [`kubectl create`](/docs/reference/generated/kubectl/kubectl-commands#create):
 
 ```shell
 $ kubectl create -f ./secret.yaml
 secret "mysecret" created
 ```
 
+For certain scenarios, you may wish to use the stringData field instead. An
+example of this is where the Secret contains a configuration file with values
+populated via templating:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+type: Opaque
+stringData:
+  config.yaml: |-
+    apiUrl: "https://my.api.com/api/v1"
+    username: {{username}}
+    password: {{password}}
+```
+
+stringData is a write-only convenience field. It is never output when
+retrieving Secrets. If a field is specified in both data and stringData, the
+value from the stringData map is used.
```

Comment:
This paragraph seems like it provides a couple of different information points that aren't necessarily related -- and so it's more confusing than it needs to be. Can you pull apart and maybe explain what's going on in the third sentence a bit more?

Reply:
@Bradamant3 hopefully that's a bit better - I've tried to give a bit more info about why you might want to use stringData, and I've tried to explain the write-only part, and the precedence between data and stringData separately.

Let me know what you think, and thanks for the feedback.